src/org/zaproxy/zap/extension/soap/WSDLCustomParser.java

```diff
@@ -489,6 +491,16 @@ private static void persistMessage(final HttpMessage message) {
 				@Override
 				public void run() {
 					extHistory.addHistory(historyRef);
+					/* Modifies the URI adding the SOAP operation name to avoid overwrites. It's done after
+					 * saving it in history so that the original URI is preserved for scanning tasks.*/
+					try {
+						URI soapURI = message.getRequestHeader().getURI();
+						String soapStringURI = soapURI.getPath();
+						soapURI.setPath(soapStringURI+opDisplayName);
```

Comment:
IMHO this is a workaround not a fix and I think it should be noted in a comment (with a ```XXX``` or ```FIXME``` tag), also it might worth raising an issue.
Since the change is being done just here it will stop working when the user reopens the session (i.e. the SOAP operations would be overwritten again).